93 함께전시 모집글 조회 기능

leejoonsang opened the pull request:
## :information_desk_person: 간단 소개
함께전시 모집글 전체/상세 조회 기능 구현

## :heavy_check_mark: 작업 내용 설명
- GatheringController, GatheringService, 그리고 Gathering 관련 데이터 리팩토링을 포함해 함께전시 모집글을 조회할 수 있는 기능 구현
- 모집글 전체/상세 조회 기능에 대한 테스트코드 작성(성공, 실패)

## :bulb: 참고사항
확인 부탁드립니다!

leejoonsang committed:
feat: 함께전시 모집글 전체/상세 조회 기능 추가
+ endpoint 수정 gathering -> gatherings

leejoonsang committed:
feat: 함께전시 모집글 전체/상세 조회 기능 추가

leejoonsang committed:
feat: 함께전시 모집글 전체/상세 조회 Response

leejoonsang committed:
feat: 필드 변수 추가

leejoonsang committed:
feat: 함께전시 모집글 전체/상세 조회 테스트 코드
1. 전체 조회 - 성공
2. 상세 조회 - 성공 / 실패(게시물 없는 경우)

leejoonsang committed:
feat: 함께전시 게시물 조회 리턴 수정 -> 테스트코드 수정
1. 전체 조회 - 성공
2. 상세 조회 - 성공 / 실패(게시물 없는 경우)

leejoonsang committed:
feat: 쿼리메소드 추가

leejoonsang committed:
feat: 필드, 메소드 수정

leejoonsang committed:
feat: 함께전시 모집글 조회 리턴 수정 -> 필드 수정, 메소드 추가

leejoonsang committed:
feat: 함께전시 모집글 조회 리턴 수정 -> 로직 수정

geuun commented on src/main/java/com/dev/museummate/service/GatheringService.java:
```diff
@@ -44,4 +51,17 @@ public GatheringDto posts(GatheringPostRequest gatheringPostRequest, String emai
         return savedDto;
     }
+
+    public Page<GatheringDto> findAllGatherings(Pageable pageable) {
```
다른 도메인 Service Layer과 동일하게

DTO 클래스로 반환하게 리펙토링 부탁드립니다 :)

geuun commented on src/main/java/com/dev/museummate/domain/dto/gathering/GatheringDto.java:
```diff
@@ -22,6 +24,15 @@ public class GatheringDto {
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    private UserEntity user;
+
```
DTO에는 Entity랑 역활 분리를 위한 클래스라서
컬럼 관계 안맺어주셔도 됩니다!

따라서 어노테이션들 지워주시면 될 것  같습니다!

geuun commented on src/main/java/com/dev/museummate/controller/GatheringController.java:
```diff
@@ -36,30 +36,28 @@ public Response<GatheringPostResponse> posts(@RequestBody GatheringPostRequest g
+                                                 .exhibitionName(gatheringResponse.getExhibitionName())
+                                                 .exhibitionMainUrl(gatheringResponse.getExhibitionMainUrl())
+                                                 .userName(gatheringResponse.getUserName())
+                                                 .createdAt(gatheringResponse.getCreatedAt())
```
Controller Layer에서 Service Layer로부터 

DTO 타입으로 받아서 Response로 반환하게 수정 부탁드립니다!

geuun commented on src/main/java/com/dev/museummate/controller/GatheringController.java:
```diff
@@ -36,30 +36,28 @@ public Response<GatheringPostResponse> posts(@RequestBody GatheringPostRequest g
     @GetMapping()
     public Response<Page<GatheringResponse>> findAllGatherings(@PageableDefault(size = 20,
             sort = "id", direction = Direction.DESC) Pageable pageable) {
-        Page<GatheringDto> gatheringDtos = gatheringService.findAllGatherings(pageable);
```
다른 도메인도 이렇게 반환하고 있어서

수정 전처럼 반환해주시면 되요..!

chlalswns200 committed:
refacto: 모집 글 조회 service에서 dto 리턴하는 방식으로 변경

Controller:
  service에서 dto까지 받아온 Controller 에서 Response 만들어서 반환
Service:
  Response 반환 -> Dto 반환으로 수정
Dto:
  연관 관계 매핑 삭제, toDto() 파라미터로 currentPeople 받도록 변경
Response:
  createGetOne() 파라미터로 currentPeople 받도록 변경

chlalswns200 committed:
test: 전체 조회 테스트 수정
Dto: 생성 시 currentPeople 추가
willReturn(): Dto 반환으로 수정

chlalswns200 commented:
말씀해주신 사항 다 반영하고 테스트 까지 완료 했습니다.
다시 확인 부탁드립니다

chlalswns200 committed:
Merge branch 'develop' into 93-함께전시-모집글-조회-기능

geuun committed:
fix: merge comflict시 발생한 버그 해결

findByPostId -> findById 변경

geuun committed:
fix: 전체적으로 Merge Comflit 중 일어난 버그 해결

GatheringResponse의 함수 명 충돌이 있던 것을
GatheringResponse / GatheringParticipantResponse로 분리

geuun reviewed (approved):
병합시 일어났던 Complict 를 해결했습니다..!

리펙토링하시느랴 고생하셨습니다!

바로 병합 진행하겠습니다!
